src/FSharpVSPowerTools.Logic/ThemeManager.fs

```diff
@@ -26,41 +31,14 @@ type ThemeManager [<ImportingConstructor>]
                (Guid("1ded0138-47ce-435e-84ef-9ec1f439b749"), VisualStudioTheme.Dark) ]
 
     let getThemeId() =
-        let dte = serviceProvider.GetService<DTE, SDTE>()
-        let version = VisualStudioVersion.fromDTEVersion dte.Version
-        match version with
-        | VisualStudioVersion.VS2012 
-        | VisualStudioVersion.VS2013 ->
-            let keyName = sprintf @"Software\Microsoft\VisualStudio\%s.0\General" (VisualStudioVersion.toString version)
-            use key = Registry.CurrentUser.OpenSubKey(keyName)
-            key.GetValue("CurrentTheme", null)
-            |> Option.ofNull
-            |> Option.map string
-        | VisualStudioVersion.VS2015 ->
-            let keyName = sprintf @"Software\Microsoft\VisualStudio\%s.0\ApplicationPrivateSettings\Microsoft\VisualStudio" 
-                            (VisualStudioVersion.toString version)
-            use key = Registry.CurrentUser.OpenSubKey(keyName)
-            key.GetValue("ColorTheme", null)
-            |> Option.ofNull
-            |> Option.map (string >> String.split StringSplitOptions.None [|"*"|])
-            |> Option.bind (function 
-                            | [|_; _; themeId|] -> Some themeId 
-                            | arr -> 
-                                Logging.logWarning (fun _ -> sprintf "Parsed Visual Studio theme settings are not well-formed %A." arr)
-                                None)
-        | _ ->
-            Logging.logWarning (fun _ -> sprintf "Can't recognize Visual Studio version %A." version)
-            None
+        let themeService = serviceProvider.GetService(typeof<SVsColorThemeService>)
+        themeService?CurrentTheme?ThemeId: Guid
```

Comment:
How robust are these dynamic calls? Could they throw exceptions in certain cases? If yes, it would be good to wrap them under `Option.attempt` just to be sure.